src/codegen/riscv64/constants-riscv64.h

```diff
@@ -849,6 +993,118 @@ class InstructionGetters : public T {
+    int32_t imm = ((Bits & 0x1800) >> 7) | ((Bits & 0x780) >> 1) 
+                  | ((Bits & 0x40) >> 4) | ((Bits & 0x20) >> 2);
+    return imm;
+  }
+
+  inline int RvcImmAddi16sp() const {
+    // | funct3 | nzimm[9] | 2 | nzimm[4|6|8:7|5] | opcode |
+    //  15         12           6                2        
+    uint32_t Bits = this->InstructionBits();
+    int32_t imm = ((Bits & 0x1000) >> 3) | ((Bits & 0x40) >> 2) 
+                  | ((Bits & 0x20) << 1) | ((Bits & 0x18) << 4) 
+                  | ((Bits & 0x4) << 3);
+    return imm << 22 >> 22;
+  }
+
+  inline int RvcImmLwsp() const {
+    // | funct3 | uimm[5] | rs1 | uimm[4:2|7:6] | opcode |
+    //  15         12            6             2        
+    uint32_t Bits = this->InstructionBits();
+    int32_t imm = ((Bits & 0x1000) >> 7) | ((Bits & 0x70) >> 2)
+                  | ((Bits & 0xc) << 4);
+    return imm;
+  }
+
+  inline int RvcImmLdsp() const {
+    // | funct3 | uimm[5] | rs1 | uimm[4:3|8:6] | opcode |
+    //  15         12            6             2        
+    uint32_t Bits = this->InstructionBits();
+    int32_t imm = ((Bits & 0x1000) >> 7) | ((Bits & 0x60) >> 2)
+                  | ((Bits & 0x1c) << 4);
+    return imm;
+  }
+
+  inline int RvcImmSwsp() const {
+    // | funct3 | uimm[5:2|7:6] | rs2 | opcode |
+    //  15       12            7                     
+    uint32_t Bits = this->InstructionBits();
+    int32_t imm = ((Bits & 0x1e00) >> 7) | ((Bits & 0x180) >> 1);
+    return imm;
+  }
+
+  inline int RvcImmSdsp() const {
+    // | funct3 | uimm[5:3|8:6] | rs2 | opcode |
+    //  15       12            7                     
+    uint32_t Bits = this->InstructionBits();
+    int32_t imm = ((Bits & 0x1c00) >> 7) | ((Bits & 0x380) >> 1);
+    return imm;
+  }
+
+  inline int RvcImmLw() const {
+    // | funct3 | uimm[5:3] | rs1′   | uimm[2|6] | opcode |
+    //  15       12       10        6         5
+    uint32_t Bits = this->InstructionBits();
+    int32_t imm = ((Bits & 0x1c00) >> 7) | ((Bits & 0x40) >> 4) 
+                  | ((Bits & 0x20) << 1);
+    return imm;
+  }
+
+  inline int RvcImmLd() const {
+    // | funct3 | uimm[5:3] | rs1′ | uimm[7:6] | rd′ | opcode |
+    //  15       12       10        6         5
+    uint32_t Bits = this->InstructionBits();
+    int32_t imm = ((Bits & 0x1c00) >> 7) | ((Bits & 0x60) << 1);
+    return imm;
+  }
+
+  inline int RvcImmJ() const {
+    // | funct3 | imm[11|4|9:8|10|6|7|3:1|5] | opcode |
+    //  15       12                         2        
+    uint32_t Bits = this->InstructionBits();
+    int32_t imm = ((Bits & 0x1000) >> 3) | ((Bits & 0x800) >> 7) 
+                  | ((Bits & 0x600) >> 1) | ((Bits & 0x100) << 2) 
+                  | ((Bits & 0x80) >> 1) | ((Bits & 0x40) << 1) 
+                  | ((Bits & 0x38) >> 2) | ((Bits & 0x4) << 3);
+    return imm << 20 >> 20;
+  }
+
+  inline int RvcImmB() const {
+    // | funct3 | imm[8|4:3] | rs1′ | imm[7:6|2:1|5] | opcode |
```

Comment:
All have the same issues as pointed out before:

1. Add DCHECK(IsShortInstruction();
2. Change the function name to include the number of bits in the immediate value, and add a `Value` postfix, e.g., `RvcImm8BValue`.
3. Name `imm` w/ the number of bits
4. For `nzimm` or `nzuimm`, check the immediate value is non-zero
5. Sign-extension for extracting signed immediate value 

Please make those fixes whenever it applies.